spot-client/src/spot-remote/ultrasound/UltrasoundTransmitter.js

```diff
@@ -0,0 +1,128 @@
+import {
+    Transmitter,
+    loadDependencies,
+    profiles,
+    str2ab
+} from 'lib-quiet-js';
+
+/**
+ * Encapsulates the implementation details loading ultrasound and transmitting
+ * ultrasound messages.
+ */
+export default class UltrasoundTransmitter {
+    /**
+     * Calls to load any other files lib-quiet-js may need.
+     *
+     * @param {string} emscriptenPath - The url path, excluding the filename,
+     * for the emscripten used by lib-quiet-js.
+     * @param {string} memoryInitializerPath - The url path, excluding the
+     * filename, for the memory file dynamically loaded by the emscripten.
+     * @returns {void}
+     */
+    static loadDependencies(emscriptenPath, memoryInitializerPath) {
+        return loadDependencies({
+            emscriptenPath,
+            memoryInitializerPath
+        });
+    }
+
+    /**
+     * Initializes a new {@code ultrasoundTransmitter} instance.
+     *
+     * @param {number} transmissionDelay - The gap between emitting ultrasonic
+     * messages.
+     * @param {string} initialText - Any text to start transmitting immediately
+     * after instantiation.
+     */
+    constructor(transmissionDelay = 3000, initialText) {
+        /**
+         * The instance plays ultrasound to transmit text and then enqueues
+         * another transmission to be transmitted after a timeout. A timeout is
+         * used to have a consistent interval between message plays.
+         *
+         * @type {timeoutID}
+         */
+        this._nextTransmissionTimeout = null;
+
+        /**
+         * The amount of time in milliseconds to wait between transmitting an
+         * ultrasound message.
+         *
+         * @type {number}
+         */
+        this._transmissionDelay = transmissionDelay;
+
+        /**
+         * The instance of {@code Transmitter} used to emit ultrasound.
+         */
+        this._transmitter = new Transmitter({
+            onFinish: this._onEnqueueNextTransmission.bind(this),
+            profile: profiles['ultrasonic-experimental']
+        });
+
+        /**
+         * The text to be transmitted using ultrasonic frequencies.
+         *
+         * @type {ArrayBuffer|null}
+         */
+        this._text = null;
+
+        this.setMessage(initialText);
+    }
+
+    /**
+     * Stops ultrasound from being emitted anymore.
+     *
+     * @returns {void}
+     */
+    destroy() {
+        this._transmitter.destroy();
+        clearTimeout(this._nextTransmissionTimeout);
+        this._text = null;
+    }
+
+    /**
+     * Updates the text to be transmitted and initializes a {@code Transmitter}
+     * to send the text if not already initialized.
+     *
+     * @param {string} text - The message to be transmitted via ultrasound.
+     * @returns {void}
+     */
+    setMessage(text = '') {
```

Comment:
Maybe this is due to the nature of the underlying library, but this API seems weird to me.  As a user, I'd expect to be able to call something like `sendMessage` and I'd expect this lib to hide away the details around doing that (retransmitting, etc.).

Reply:
Is that not what happens now. UltrasoundTransmitter hides away the retransmitting.

Reply:
`setMessage` is confusing to me as a user, though.  I want to _send_ a message.  I'd think I need to call something else after `setMessage` to actually send it.  Underneath, you're setting a message and then something is checking it, but that's just confusing as a user I think.  As far as the user is concerned this is sending, so just call it `sendMessage`.